### PR TITLE
Np 882 orgunitid as uri

### PR DESCRIPTION
**Pull request by nraanes**
The main focus of this PR should be:
AddNewAuthorityIdentifierHandler
DeleteAuthorityIdentifierHandler
UpdateAuthorityIdentifierHandler

Changes:
Identifier (previously path param) moved to request body
Upgrade from junit4->5, mockito2->3, removed  dep javax.ws.rs-api 
Using nva-commons and nva-testutils
No longer using gson (for the mentioned handlers)
Improved testnaming (for the mentioned handlers)
Template updated

Unfortunately due to changes in dependencies most other classes also have some minor changes.

### CODE REVIEW

File: src/main/java/no/unit/nva/bare/AddNewAuthorityIdentifierHandler.java [hunk 1]
```diff
@@ -14,63 +17,59 @@
...
-        }
-        Map<String, String> pathParameters = (Map<String, String>) input.get(PATH_PARAMETERS_KEY);
-        String scn = pathParameters.get(SCN_KEY);
-        String inputQualifier = pathParameters.get(QUALIFIER_KEY);
+    protected Authority processInput(AddNewAuthorityIdentifierRequest input, RequestInfo requestInfo,
+                                     Context context) throws ApiGatewayException {
+
+        validateInput(input, requestInfo.getPathParameters());
+
+        String scn = requestInfo.getPathParameters().get(SCN_KEY);
```
Comment (janbj) [inline src/main/java/no/unit/nva/bare/AddNewAuthorityIdentifierHandler.java:69]: RequestUtil.getPathParameter(requestInfo, SCN_KEY)

Håndterer feil for deg.

File: src/main/java/no/unit/nva/bare/AddNewAuthorityIdentifierHandler.java [hunk 2]
```diff
@@ -82,62 +81,58 @@ private String transformQualifier(String inputQualifier) {
...
+            throw new InvalidInputException(MISSING_REQUEST_JSON_BODY);
+        }
+        if (StringUtils.isEmpty(input.getIdentifier())) {
+            throw new InvalidInputException(MISSING_ATTRIBUTE_IDENTIFIER);
         }
     }
 
-    protected GatewayResponse addNewIdentifier(String scn, String qualifier, String identifier) {
-        GatewayResponse gatewayResponse = new GatewayResponse();
+    protected Authority addNewIdentifier(String scn, String qualifier, String identifier)
```
Comment (janbj) [inline src/main/java/no/unit/nva/bare/AddNewAuthorityIdentifierHandler.java:103]: Splitt denne i flere metoder, det er altfor mye nøsting her.

File: src/main/java/no/unit/nva/bare/Authority.java [hunk 3]
```diff
@@ -1,18 +1,34 @@
...
 
     public static final String EMPTY_STRING = "";
+
+    @JsonProperty("name")
     private String name;
+
+    @JsonProperty("systemControlNumber")
     private String systemControlNumber;
+
+    @JsonProperty("feideids")
```
Comment (janbj) [inline src/main/java/no/unit/nva/bare/Authority.java:19]: Jeg hadde forventet feideIds

File: src/main/java/no/unit/nva/bare/BareCommunicationException.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package no.unit.nva.bare;
+
+import nva.commons.exceptions.ApiGatewayException;
+import org.apache.http.HttpStatus;
+
+public class BareCommunicationException extends ApiGatewayException {
+
+    public static final int ERROR_CODE = HttpStatus.SC_INTERNAL_SERVER_ERROR;
```
Comment (janbj) [inline src/main/java/no/unit/nva/bare/BareCommunicationException.java:8]: Er det egentlig en internal server error når ekstern tjeneste gir feil?

File: src/main/java/no/unit/nva/bare/CreateAuthorityHandler.java [hunk 5]
```diff
@@ -66,28 +70,28 @@ protected GatewayResponse createAuthorityOnBare(String name) {
         try {
             HttpResponse<String> response = bareConnection.createAuthority(bareAuthority);
```
Comment (janbj) [inline src/main/java/no/unit/nva/bare/CreateAuthorityHandler.java:71]: Denne må splittes opp, altfor mye som skjer i en metode og altfor mye nøsting.

File: src/main/java/no/unit/nva/bare/DeleteAuthorityIdentifierHandler.java [hunk 6]
```diff
@@ -14,133 +17,124 @@
...
-        }
-        Map<String, String> pathParameters = (Map<String, String>) input.get(PATH_PARAMETERS_KEY);
-        String scn = pathParameters.get(SCN_KEY);
-        String inputQualifier = pathParameters.get(QUALIFIER_KEY);
+    protected Authority processInput(DeleteAuthorityIdentifierRequest input, RequestInfo requestInfo,
+                                     Context context) throws ApiGatewayException {
+
+        validateInput(input, requestInfo.getPathParameters());
+
+        String scn = requestInfo.getPathParameters().get(SCN_KEY);
```
Comment (janbj) [inline src/main/java/no/unit/nva/bare/DeleteAuthorityIdentifierHandler.java:69]: RequestUtil håndterer feil for deg.

File: src/main/java/no/unit/nva/bare/DeleteAuthorityIdentifierHandler.java [hunk 7]
```diff
@@ -14,133 +17,124 @@
...
+            throw new InvalidInputException(MISSING_REQUEST_JSON_BODY);
+        }
+        if (StringUtils.isEmpty(input.getIdentifier())) {
+            throw new InvalidInputException(MISSING_ATTRIBUTE_IDENTIFIER);
         }
     }
 
-    protected GatewayResponse deleteIdentifier(String scn, String qualifier, String identifier) {
-        GatewayResponse gatewayResponse = new GatewayResponse();
+    protected Authority deleteIdentifier(String scn, String qualifier, String identifier)
```
Comment (janbj) [inline src/main/java/no/unit/nva/bare/DeleteAuthorityIdentifierHandler.java:104]: Splitt opp

File: src/main/java/no/unit/nva/bare/FetchAuthorityHandler.java [hunk 8]
```diff
@@ -58,11 +61,11 @@ public GatewayResponse handleRequest(final Map<String, Object> input, Context co
                     BareAuthority fetchedAuthority = bareConnection.get(arpId);
```
Comment (janbj) [inline src/main/java/no/unit/nva/bare/FetchAuthorityHandler.java:61]: Splitt opp

File: src/main/java/no/unit/nva/bare/AddNewAuthorityIdentifierHandler.java (hunk 1 above)
Comment (nraanes) [reply src/main/java/no/unit/nva/bare/AddNewAuthorityIdentifierHandler.java:71]: Nice, I didnt notice yesterdays new version, will bump commons to 0.3.5 and make use of it!

File: src/main/java/no/unit/nva/bare/AddNewAuthorityIdentifierHandler.java [hunk 9]
```diff
@@ -14,63 +17,59 @@
...
-            "Communication failure while updating authority %s";
-    public static final String PATH_PARAMETERS_KEY = "pathParameters";
+            "Communication failure trying to update authority";
     public static final String SCN_KEY = "scn";
     public static final String QUALIFIER_KEY = "qualifier";
-    public static final String IDENTIFIER_KEY = "identifier";
-    public static final int ERROR_CALLING_REMOTE_SERVER = Response.Status.BAD_GATEWAY.getStatusCode();
     public static final String REMOTE_SERVER_ERRORMESSAGE = "remote server errormessage: ";
 
     public static final List<String> VALID_QUALIFIERS = asList(ValidIdentifierKey.FEIDEID.asString(),
```
[the remark this answers is not in the document]
Comment (nraanes) [reply src/main/java/no/unit/nva/bare/AddNewAuthorityIdentifierHandler.java]: Should be fixed now. Also, the list is repeated in the three handlers, but that might come in handy if it is determined there are differences in which identifiers one is allowed to add vs delete vs update.

File: src/main/java/no/unit/nva/bare/Authority.java (hunk 3 above)
Comment (nraanes) [reply src/main/java/no/unit/nva/bare/Authority.java:19]: I didn't want to change the existing object returned to frontend, but I will check what is actually returned as of now before merging my change.

File: src/main/java/no/unit/nva/bare/CreateAuthorityHandler.java (hunk 5 above)
Comment (nraanes) [reply src/main/java/no/unit/nva/bare/CreateAuthorityHandler.java:71]: I have created a task for upgrading this handler (outside scope of task at hand, but changes in dependencies forced a few changes nonetheless)

https://unit.atlassian.net/browse/NP-1063

File: src/main/java/no/unit/nva/bare/FetchAuthorityHandler.java (hunk 8 above)
Comment (nraanes) [reply src/main/java/no/unit/nva/bare/FetchAuthorityHandler.java:61]: I have created a task for upgrading this handler (outside scope of task at hand, but changes in dependencies forced a few changes nonetheless)

https://unit.atlassian.net/browse/NP-1063

File: build.gradle [hunk 10]
```diff
@@ -13,36 +12,40 @@ sourceCompatibility = JavaVersion.VERSION_11  // source-code version and must be
...
-        archiveClassifier.set('')
-        zip64 true
-    }
+    implementation group: 'org.zalando', name: 'problem', version: zalandoVersion
+    implementation group: 'org.zalando', name: 'jackson-datatype-problem', version: zalandoVersion
+
+    testImplementation group: 'com.github.BIBSYSDEV', name: 'nva-testutils', version: 'v0.1.7'
+    testImplementation group: 'org.hamcrest', name: 'hamcrest', version: '2.2'
+    testImplementation group: 'org.mockito', name: 'mockito-core', version: '3.2.4'
+    testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-engine', version: junit5Version
```
Comment (janbj) [inline build.gradle:36]: testImplementation og testRuntimeOnly?

File: src/main/java/no/unit/nva/bare/AddNewAuthorityIdentifierHandler.java (hunk 1 above)
Comment (svenngjeto) [reply src/main/java/no/unit/nva/bare/AddNewAuthorityIdentifierHandler.java]: Using v0.3.5 and getPAthParameter

File: build.gradle (hunk 10 above)
Comment (svenngjeto) [reply build.gradle:36]: removed double dependency

File: src/main/java/no/unit/nva/bare/DeleteAuthorityIdentifierHandler.java (hunk 6 above)
Comment (svenngjeto) [reply src/main/java/no/unit/nva/bare/DeleteAuthorityIdentifierHandler.java]: Jess

**Pull request merged**